crates/ty_python_semantic/resources/mdtest/typed_dict.md

```diff
@@ -169,6 +173,57 @@ def _(person: Person, literal_key: Literal["age"], union_of_keys: Literal["age",
     reveal_type(person[unknown_key])  # revealed: Unknown
 ```
 
+### Writing
+
+```py
+from typing_extensions import TypedDict, Final, Literal, LiteralString, Any
+
+class Person(TypedDict):
+    name: str
+    surname: str
+    age: int | None
+
+NAME_FINAL: Final = "name"
+AGE_FINAL: Final[Literal["age"]] = "age"
+
+def _(person: Person):
+    person["name"] = "Alice"
+    person["age"] = 30
+
+    # error: [invalid-key] "Invalid key access on TypedDict `Person`: Unknown key "naem" - did you mean "name"?"
+    person["naem"] = "Alice"
+
+def _(person: Person):
+    person[NAME_FINAL] = "Alice"
+    person[AGE_FINAL] = 30
+
+def _(person: Person, literal_key: Literal["age"]):
+    person[literal_key] = 22
+
+def _(person: Person, union_of_keys: Literal["name", "surname"]):
+    person[union_of_keys] = "unknown"
+
+    # error: [invalid-assignment] "Cannot assign value of type `Literal[1]` to key of type `Literal["name", "surname"]` on TypedDict `Person`"
```

Comment:
I feel like ideally we would tell you which key(s) actually failed to assign, if only some of them do. But I see that that would add significant complexity to the implementation (we'd have to dig into the CallError details), so I'm fine not doing it for now and seeing if it's an issue in practice